File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-shared-5-4/Step.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class Step : StepBase
    {
        public Step(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode stepValueNode = xmlNode.SelectSingleNode("stepValue");
            
            if (stepValueNode != null)
            {
                if (stepValueNode.Attributes["href"] != null || stepValueNode.Attributes["id"] != null) 
                {
                    if (stepValueNode.Attributes["id"] != null) 
                    {
                        stepValueIDRef_ = stepValueNode.Attributes["id"].Value;
                        XsdTypeDecimal ob = new XsdTypeDecimal(stepValueNode);
                        IDManager.SetID(stepValueIDRef_, ob);
                    }
                    else if (stepValueNode.Attributes["href"] != null)
                    {
                        stepValueIDRef_ = stepValueNode.Attributes["href"].Value;
                    }
                    else
                    {
                        stepValue_ = new XsdTypeDecimal(stepValueNode);
                    }
                }
                else
                {
                    stepValue_ = new XsdTypeDecimal(stepValueNode);
                }
            }
            
        
        }
        
    
        #region StepValue_
        private XsdTypeDecimal stepValue_;
        public XsdTypeDecimal StepValue_
        {
            get
            {
                if (this.stepValue_ != null)
                {
                    return this.stepValue_; 
                }
                else if (this.stepValueIDRef_ != null)
                {
                    stepValue_ = IDManager.getID(stepValueIDRef_) as XsdTypeDecimal;
                    return this.stepValue_; 
                }
                else
                {
                      return this.stepValue_; 
                }
            }
            set
            {
                if (this.stepValue_ != value)
                {
                    this.stepValue_ = value;
                }
            }
        }
        #endregion
        
        public string stepValueIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-shared-5-4/Step.cpp
// Step.cpp 
#include "Step.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

Step::Step(TiXmlNode* xmlNode)
: StepBase(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //stepValueNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* stepValueNode = xmlNode->FirstChildElement("stepValue");

   if(stepValueNode){stepValueIsNull_ = false;}
   else{stepValueIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- stepValueNode , address : " << stepValueNode << std::endl;
   #endif
   if(stepValueNode)
   {
      if(stepValueNode->Attribute("href") || stepValueNode->Attribute("id"))
      {
          if(stepValueNode->Attribute("id"))
          {
             stepValueIDRef_ = stepValueNode->Attribute("id");
             stepValue_ = boost::shared_ptr<XsdTypeDecimal>(new XsdTypeDecimal(stepValueNode));
             stepValue_->setID(stepValueIDRef_);
             IDManager::instance().setID(stepValueIDRef_,stepValue_);
          }
          else if(stepValueNode->Attribute("href")) { stepValueIDRef_ = stepValueNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { stepValue_ = boost::shared_ptr<XsdTypeDecimal>(new XsdTypeDecimal(stepValueNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<XsdTypeDecimal> Step::getStepValue()
{
   if(!this->stepValueIsNull_){
        if(stepValueIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeDecimal>(IDManager::instance().getID(stepValueIDRef_));
        }else{
             return this->stepValue_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDecimal>();
   }
}
}

